internal/validator/default/helpers.go

```diff
@@ -0,0 +1,78 @@
+package defaultvalidator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/bitcoin-sv/arc/internal/validator"
+	"github.com/libsv/go-bt/v2"
+)
+
+var errParentNotFound = errors.New("parent transaction not found")
+
+func extendTx(ctx context.Context, w validator.TxFinderI, rawTx *bt.Tx) error {
+	// potential improvement: implement version for the rawTx with only one input
+
+	// get distinct parents
+	// map parentID with inputs collection to avoid duplication and simplify later processing
+	parentInputMap := make(map[string][]*bt.Input)
+	parentsIDs := make([]string, 0, len(rawTx.Inputs))
+
+	for _, in := range rawTx.Inputs {
+		prevTxID := in.PreviousTxIDStr()
+
+		inputs, found := parentInputMap[prevTxID]
+		if !found {
+			// first occurrence of the parent
+			inputs = make([]*bt.Input, 0)
+			parentsIDs = append(parentsIDs, prevTxID)
+		}
+
+		inputs = append(inputs, in)
+		parentInputMap[prevTxID] = inputs
+	}
+
+	// get parents
+	parentsTxs, err := w.GetRawTxs(ctx, parentsIDs)
+	if err != nil {
+		return fmt.Errorf("cannot extend tx: %w", err)
```

Comment:
I think the error message is not correct
Maybe this "Failed to get raw transactions for parent IDs: %w" ?